src/rpc/net.cpp

```diff
@@ -863,15 +863,11 @@ static RPCHelpMan getnodeaddresses()
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
     }
 
-    int count = 1;
-    if (!request.params[0].isNull()) {
-        count = request.params[0].get_int();
-        if (count < 0) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
-        }
-    }
+    const int count{request.params[0].isNull() ? 1 : request.params[0].get_int()};
```

Comment:
1f24a27a4c80f182b275a75afdb4777ec792fd91

FWIW #20017 makes this more simple.

Reply:
TIL about that pull, thanks! (One line as done here is pretty simple though?) Resolving as not actionable ATM in this patch.